File: C/SEM I/Assignment III/5.c
/* 

Assignemnt III
Name: Adersh Varshnei
5. Write a C Program to find all prime factors of a number

*/

#include <stdio.h>
#include <conio.h>
#include <stdlib.h>
#include <math.h>

//************************************************************************************************

typedef unsigned int loopVar;

//************************************************************************************************

void primeFactFind(int);

//************************************************************************************************

int count = 0;

//************************************************************************************************

int main(void) {
	loopVar rep = 1;

	while (rep == 1) {
		int inputNum;

		printf("Enter the number whose prime factors are to be found (Positive Factors will be found for negative numbers): ");
		scanf("%d", & inputNum);

		primeFactFind(inputNum);

		printf("\nRepeat Program?: ");
		scanf("%d", & rep);

		count++;

		if (count >= 15) {
			printf("Abnormal END\n");

			getch();
			exit(0);
		} //if (count >= 15) 
	} //while (rep == 1) 

	printf("Normal END\n");

	getch();
	return 0;
} //int main(void)

//************************************************************************************************

void primeFactFind(int num) {
	loopVar i = 2;
	int temp = num;

	printf("The prime factors of %d are:\n", num);

	do {
		if (abs(temp) % i == 0) {
			printf("%d\t", i);

			temp /= i;
			i = 2;
		} else {
			i++;
		} //End of extended if clause
	} while (i <= abs(temp));
} //void primeFactFind(int num)

File: Python/assignment iii/5.py
# Program to find prime factors

import math

def primeChk(num):
    flag = True
    
    for i in range(2, int(math.sqrt(num))):
        if num % i == 0:
            flag == False
            break
    
    return flag

def primeFactors(num):
    i = 2
    fact = []

    while num > 1:
        if (num % i == 0) and (primeChk(i) == True):
            fact.append(i)
            num = int(num / i)
            i = 2
        else:
            i += 1
    
    return fact

def printList(entList):
    for i in range(len(entList)):
        print(f"{entList[i]}", end = ' ')

num = int(input("Enter the number: "))

print(f"Prime factors of {num} are: ", end = '')
printList(primeFactors(num))